### PR TITLE
Update Rails framework to 5.2

### PR DESCRIPTION
**Pull request by ZeroPointEnergy**
Over the last couple of weeks I incrementally upgraded the rails framework minor release for minor release until the latest 5.2 release.

The tests are all passing for MySQL and Postgres and all ruby releases from 2.2 -> 2.5. However there are for sure parts of the application which have no test coverage and it may break stuff (IMHO the custom rake tasks).

Over the next weeks I will deploy and test this in a production-like environment and I'm sure there will be more changes coming to the PR.

The reason why I opened this yet and not when it is finished is that I would be glad to get some early reviews if someone is interested in helping out . And I also feel that we probably need to talk about the future of this project and the best way to do this is always to dump some heavy code changes on the maintainer :wink: 

Let me know what you think and how you see the way forward for the puppet-dashboard project.

### CODE REVIEW

**Comment by sodabrew** [comment]
This is fantastic, thank you!

**Comment by ZeroPointEnergy** [comment]
I have this running in production now (with ruby 2.4 and passenger) for a while and it looks like everything works as it should.

I also tried to fix the remaining tasks which have no test coverage and tested them by hand for now.

From my side I think this is ready but would be great if someone else could test it. If you have any concerns let me know.

File: .travis.yml [hunk 1]
```diff
@@ -6,15 +6,15 @@ services:
...
   - postgresql
 rvm:
-  - 1.8.7
-  - 1.9.3
-  - 2.0.0
-  - 2.1
+  - 2.2
+  - 2.3
+  - 2.4
+  - 2.5
```
Comment (smortex) [inline .travis.yml]: You may also want to update the README accordingly :wink:

Comment (bwitt) [reply .travis.yml]: @ZeroPointEnergy would you mind making a PR for those fixes too?  I have a feeling we can get them merged now.

**Pull request merged**